### ISSUE TITLE
Discussion: Tree selection (nested detail/summary with checkboxes)

### ISSUE THREAD

**Issue by juleskuehn** [wet-boew/wet-boew#8666]
Just curious: Has anyone implemented a selection tree in WET? Each `<detail>` element corresponds to an inner node (with children). Leaf nodes are explicit `<div class="leaf_node">`.

It seems to me that the [menu plugin](https://wet-boew.github.io/wet-boew/unmin/demos/menu/menu-en.html) is closest to this goal in terms of arrow key behaviour, although the spacebar will have to be reallocated to selecting or deselecting the checkbox.

So far, I have been simply using nested detail/summary elements with checkboxes inside. This is posing some accessibility issues which I can work through, but I don't want to do so unnecessarily if this work has already been done.

Simplified HTML:
```
<div class="selection_tree">
  <details open="">
    <summary>
      <div class="checkbox">
        <label>
          <input type="checkbox">
          1 selectable item
        </label>
      </div>
    </summary>
    <details open="">
      <summary>
        <div class="checkbox">
          <label>
            <input type="checkbox">
            1.1 selectable sub-item
          </label>
        </div>
      </summary>
      <div class="leaf_node">
        <div class="checkbox">
          <label>
            <input type="checkbox">
            1.1.1 selectable sub-sub-item
          </label>
        </div>
      </div>
...
```
Some JavaScript will allow for selections of parents to cascade to children and vice-versa, depending on desired behaviour.

**Comment by duboisp** [comment]
There is nothing currently in WET that support tree selection.

You can find some information about the Tree view in the [ARIA 1.1. Practices](https://www.w3.org/TR/wai-aria-practices-1.1/#TreeView) which is accompanied by 4 working example. Or [the Treegrid](https://www.w3.org/TR/wai-aria-practices-1.1/#treegrid) accompanied by 1 example

**Comment by juleskuehn** [comment]
Indeed, I was planning to model the interaction based on the ARIA tree view. This does not include any specification for the behaviour of checkboxes.

There is an additional issue of presenting the "indeterminate" state of checkboxes (available via Javascript) non-visually. A checkbox would be "indeterminate" when some, but not all, of its children are checked.

One of the best examples I could find is from the [Wijmo framework TreeView.](https://demos.wijmo.com/5/PureJS/TreeViewIntro/TreeViewIntro/) but I have not tested for accessibility.

In any case, I guess this is a candidate for a WET plugin. I will work on it.

**Comment by duboisp** [comment]
I suggest to documents all your research, your use case, your prototype... Let me know if you are interested and I will create you a dedicated research page for your project in the wet-boew documentation website here: https://wet-boew.github.io/wet-boew-documentation/research-en.html
Don't hesitate to join us during the weekly Tuesday code sprint.

Here some information in regards of the "indeterminate" state:

Checkbox already have an indeterminate attribute and it is already supported. ref. https://html.spec.whatwg.org/multipage/input.html#checkbox-state-(type=checkbox)

HTML5 spec noted that the checkbox is never a true tri-state control. The indeterminate attribute only gives the appearance of a third state.

The WAI-ARIA 1.1 define 4 different checked state value for a checkbox controls. The "mixed" value indicate the value for a tri-state checkbox. ref. https://www.w3.org/TR/wai-aria-1.1/#aria-checked

There is an example here of a tri-state checkbox in [wai-aria practices 1.1](https://www.w3.org/TR/wai-aria-practices-1.1/#checkbox) here: https://www.w3.org/TR/wai-aria-practices-1.1/examples/checkbox/checkbox-2/checkbox-2.html

**Comment by juleskuehn** [comment]
OK, I will do my best to create some documentation as I go.

I prefer using the native `input type="checkbox"` elements with indeterminate attribute rather than a `div role="checkbox"`. As seen in the ARIA example you linked, the mixture of native and non-native checkbox types creates styling difficulties. My particular application requires that parent elements are true `input type="checkbox"` and have a `checked `attribute when any children are `checked`. I will follow the ARIA pattern for cycling between 3 states though (all checked, all unchecked, last state), and the ARIA tree view pattern.

Hopefully we can discuss this further at the code sprint on July 9th.
